<?xml version="1.0"?>
<launch>

<arg name="robot_name" default="erebor"/>
<arg name="robot_base_frame" default="base_footprint"/>
<arg name="map_frame" default="/map"/>
<arg name="use_gazebo" default="false"/>

<arg name="urdf_file" default="$(find era_gazebo)/urdf/simple_box.urdf" />

<param name="simple_box_description" textfile="$(arg urdf_file)" />

<node pkg="era_gazebo" name="ERAmsgBuilder_node" type="ERAmsgBuilder_node" >
  <param name="ID" value="$(arg robot_name)"/>
  <param name="map_frame" value="$(arg map_frame)"/>
  <param name="robot_base_frame" value="$(arg robot_base_frame)"/>
</node>

<node pkg="era_gazebo" name="ERAmsgInterpreter_node" type="ERAmsgInterpreter_node" >
  <param name="use_gazebo" value="$(arg use_gazebo)"/>
</node>

<arg name="multicast_ip" default="9.2.194.176" />
<arg name="multicast_port" default="52001" />

<node pkg="dsrc" name="wifi_transceiver"
    launch-prefix=""
    type="wifi_transceiver.py"
    args="--multicast-ip $(arg multicast_ip)
          --robot-name $(arg robot_name)
          --multicast-port $(arg multicast_port)
          --encoding 0
          --freq 5890000000
          --samp-rate 10e6
          --tx-gain 0.75
          --rx-gain 0.75
          --chan-est 0
          --lo-offset 0"
    output="screen"/>


<node pkg="era_gazebo" name="map_fuser" type="map_fuser">
  <param name="max_distance" value="100.0"/>
</node>

</launch>

